dozer/cogs/info.py

```diff
@@ -10,67 +12,119 @@
+        embed.add_field(name='Color', value=str(member.color).upper(), inline=True)
+
+        status = 'DND' if member.status is discord.Status.dnd else member.status.name.title()
+        if member.status is not discord.Status.offline:
+            platforms = self.pluralize([platform for platform in ('web', 'desktop', 'mobile') if
+                                        getattr(member, f'{platform}_status') is not discord.Status.offline])
+            status = f'{status} on {platforms}'
+        activities = '\n'.join(self._format_activities(member.activities))
+        embed.add_field(name='Status and Activity', value=f'{status}\n{activities}', inline=True)
+
+        embed.add_field(name='Roles', value=', '.join(role.name for role in member.roles[:0:-1]) or 'None')
+        embed.add_field(name='Icon URL', value=icon_url)
+        embed.set_thumbnail(url=icon_url)
+        await ctx.send(embed=embed)
 
     member.example_usage = """
-    `{prefix}member` - get information about yourself
-    `{prefix}member cooldude#1234` - get information about cooldude
+    `{prefix}member`: show your member info
+    `{prefix}member {ctx.me}`: show my member info
     """
 
+    @staticmethod
+    def _format_activities(activities: typing.Sequence[discord.Activity]) -> typing.List[str]:
+        if not activities:
+            return []
+
+        def format_activity(activity: discord.Activity) -> str:
+            if isinstance(activity, discord.CustomActivity):
+                return f"{activity.emoji} {activity.name}"
+            elif isinstance(activity, discord.Spotify):
+                return f'Listening to {activity.title} by {activity.artist} on Spotify'
+            elif activity.type is discord.ActivityType.listening:
+                return f'Listening to {activity.name}'  # Special-cased to insert " to"
+            else:
+                return f'{activity.type.name.capitalize()} {activity.name}'
+
+        # Some games show up twice in the list (e.g. "Rainbow Six Siege" and "Tom Clancy's Rainbow Six Siege") so we
+        # need to dedup them by string similarity before displaying them
+        matcher = SequenceMatcher(lambda c: not c.isalnum(), autojunk=False)
+        filtered = [activities[0]]
+        for activity in activities[1:]:
+            matcher.set_seq2(activity.name)  # Expensive metadata is computed about seq2, so change it less frequently
+            for filtered_activity in filtered:
+                matcher.set_seq1(filtered_activity.name)
+                if matcher.quick_ratio() < 0.6 and matcher.ratio() < 0.6:  # Use quick_ratio if we can as ratio is slow
+                    filtered.append(activity)
+                    break
+
+        return [format_activity(activity) for activity in filtered]
+
+    @staticmethod
+    def pluralize(values: typing.List[str]) -> str:
+        """Given a list of items, it inserts commas and "and"s in the right places to create a grammatically correct list."""
```

Comment:
Copying and pasting my comment into a docstring isn't what I meant. 😂 Please rephrase this to fill in the blank in "What does this function do? It ___" as most of our other function docs do.